### PR TITLE
Fix handling of `ltcmd` newlines (`\obeyedline`) in l3doc

### LINKED ISSUE #1577: l3doc: `\obeyedline` found in function list
The current handling of `\obeyedline` in `l3doc` (see https://github.com/latex3/latex3/commit/bb30f673440bfa4f24746a92aea3f02e306f62ee which was picked from https://github.com/latex3/latex2e/commit/9a544750a4b4a1e822ded0c73e12a7c79d041e8b) only works for `.dtx` files where doc lines all begin with `%`.

In a simple `.tex` file loading `l3doc`, when the mandatory argument of `function` or `macro` envs consists of newlines, `\obeyedline` along with space characters at the beginning of lines can be found in the typeset function list.

These `\obeyedline` are new representation of end-of-line, produced by `+v` `ltcmd` argument specifier since LaTeX 2024-06-01.

```tex
\documentclass{l3doc}

\begin{document}
\begin{function}{\my_fun:}
  content
\end{function}

\begin{function}
  {
    \my_fun:
  }
  content
\end{function}
\end{document}
```

image

I'll provide a PR.

### PR DESCRIPTION
**Pull request by muzimuzhi**
Fixes #1577.

### CODE REVIEW

File: l3kernel/l3doc.dtx [hunk 1]
```diff
@@ -2172,17 +2180,20 @@ and all files in that bundle must be distributed together.
...
       }
       {
+        \tl_set:Nn \l_@@_tmpa_tl {#1}
         \tl_remove_all:Ne \l_@@_tmpa_tl
-          { \iow_char:N \^^M \c_percent_str }
+          { \exp_not:N \obeyedline \c_percent_str }
         \tl_remove_all:Ne \l_@@_tmpa_tl
-          { \token_to_str:N \obeyedline \c_space_tl \c_percent_str }
+          { \exp_not:N \obeyedline }
+        \__kernel_tl_set:Nx \l_@@_tmpa_tl { \tl_to_str:N \l_@@_tmpa_tl }
```
Comment (muzimuzhi) [inline l3kernel/l3doc.dtx:2194]: a long code line

File: l3kernel/l3doc.dtx [hunk 2]
```diff
@@ -2172,17 +2180,20 @@ and all files in that bundle must be distributed together.
 %    \begin{macrocode}
 \cs_new_protected:Npn \@@_names_get_seq:nN #1#2
   {
-    \__kernel_tl_set:Nx \l_@@_tmpa_tl { \tl_to_str:n {#1} }
     \bool_if:NTF \l_@@_names_verb_bool
       {
         \seq_clear:N #2
-        \seq_put_right:NV #2 \l_@@_tmpa_tl
+        \seq_put_right:Ne #2 { \tl_to_str:n {#1} }
```
Comment (muzimuzhi) [inline l3kernel/l3doc.dtx:2186]: Would `:No` be quicker?

Comment (FrankMittelbach) [reply l3kernel/l3doc.dtx:2186]: I would think so (and in some sense more accurate in my opinion)

Comment (muzimuzhi) [reply l3kernel/l3doc.dtx:2186]: And lucky the doc of `\tl_to_str:n` says "The base function requires only a single expansion."

**Pull request merged**